.github/workflows/continuous_delivery.yml

```diff
@@ -0,0 +1,58 @@
+name: Publish and Tag
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "pyproject.toml"
+
+jobs:
+  check_version:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Install Poetry
+        run: |
+          curl -sSL https://install.python-poetry.org | python3 -
+          export PATH="$HOME/.poetry/bin:$PATH"
+
+      - name: Install project dependencies with Poetry
+        run: |
+          poetry install
+
+      - name: Extract version from pyproject.toml
+        id: extract_version
+        run: |
+          VERSION=$(poetry version -s)
+          echo "PACKAGE_VERSION=${VERSION}" >> $GITHUB_ENV
+
+      - name: Verify if version has changed
+        id: version_check
+        run: |
+          if [ $(git tag -l "v${{ env.PACKAGE_VERSION }}") ]; then
+            echo "Version ${{ env.PACKAGE_VERSION }} already exists."
+            exit 0
+          fi
+
```

Comment:
Please add all the style checks and unit tests execution as done in continuous_integration.yml

Reply:
Done.
I also added "main" in CI config's PR triggers, since the CD only covers for changes in push.
I removed the dependency on changes in pyproject.toml, as this is already covered by the fi...end action under "Verify if version has changed"